unlock-app/src/services/storageService.js

```diff
@@ -371,4 +373,37 @@ export class StorageService extends EventEmitter {
       this.emit(failure.ejectUser, { publicKey })
     }
   }
+
+  /*
+   * Given a lock address, a key ID, and a typed data signature, get
+   * the metadata (public and protected) associated with that key.
+   * @param {string} lockAddress
+   * @param {string} keyId
+   * @param {*} signature
+   * @param {*} data
+   */
+  async getMetadataFor(lockAddress, keyId, signature, data) {
+    const opts = {
+      headers: this.genAuthorizationHeader(signature),
+    }
+    const stringData = encodeURIComponent(JSON.stringify(data))
+    try {
+      const result = await axios.get(
+        `${this.host}/api/key/${lockAddress}/${keyId}?data=${stringData}&signature=${signature}`,
```

Comment:
there may be a way of building the query string using whatever axios provides, this is not a blocker but may make this slightly cleaner